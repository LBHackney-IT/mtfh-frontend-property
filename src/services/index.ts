export * from "./config";
export { default as locale } from "./locale";
