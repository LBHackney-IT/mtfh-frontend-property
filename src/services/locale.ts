import { AssetAddress, AssetType } from "@mtfh/common/lib/api/asset/v1";

export default {
  backToSearch: "Search results",
  cancel: "Cancel",
  noDataForKey: "N/A",
  assets: {
    assetDetails: {
      address: (assetAddress: AssetAddress): string => {
        const {
          postPreamble,
          addressLine1,
          addressLine2,
          addressLine3,
          addressLine4,
          postCode,
        } = assetAddress;
  
        return [
          postPreamble,
          addressLine1,
          addressLine2,
          addressLine3,
          addressLine4,
          postCode,
        ]
          .filter((addressLine) => !!addressLine)
          .join(" ");
      },
      newTenure: "New tenure",
    },
    assetType: (assetType: AssetType): string => {
      if (assetType === "Dwelling") {
        return "Dwelling";
      }
      if (assetType === "LettableNonDwelling") {
        return "Lettable non-dwelling";
      }
      return assetType;
    },
    patchAssetAddressSuccessMessage: "The asset address has been updated successfully."
  },
  comments: {
    heading: "Comments",
    addComment: "Add comment",
  },
  repairs: {
    heading: "Repairs",
  },
  tenureDetails: {
    tenureLabel: "Tenure",
    expandedTenureSection: "Tenure details",
    viewTenureButtonLabel: "View tenure",
    assetTypeLabel: "Type",
    assetPaymentReferenceLabel: "Payment reference",
    uprnLabel: "UPRN",
    assetReferenceLabel: "Reference",
    active: "Active",
    isActive: (isActive: boolean): string => (isActive ? "Active" : "Inactive"),
    status: "Status",
    type: "Type",
    startDate: "Start date",
    endDate: "End date",
    noTenure: "No tenure",
    paymentRef: "Payment ref.",
  },
  cautionaryAlerts: {
    cautionaryAlerts: "Cautionary alerts",
    none: "None",
  },
  static: {
    newProcess: "New Process",
  },
  assetCouldNotBeLoaded: "This property could not be loaded.",
  errors: {
    unableToFetchRecord: "There was a problem retrieving the record",
    assetDoesNotExist: "The property information you've requested does not exist",
    unableToFetchRecordDescription:
      "Please try again. If the issue persists, please contact support.",
    alertTitle: "Error",
    postcodeInvalid: "Please enter a valid postcode",
    unableToPatchAsset: "There was a problem amending this asset",
    unableToPatchAssetInvalidVersion: "The 'version' property is invalid on this asset. This is a required property when updating the asset.",
    unableToPatchAssetDescription:
      "Please refresh the page and try again. If the issue persists, please contact support.",
  },
};
