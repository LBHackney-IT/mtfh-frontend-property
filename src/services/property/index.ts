export * from './property';
export * from './property.types';
