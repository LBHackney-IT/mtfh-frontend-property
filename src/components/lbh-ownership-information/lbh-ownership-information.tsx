import React from "react";
import { Link as RouterLink } from "react-router-dom";

import { locale } from "../../services";
import { assetAdminAuthGroups } from "../../services/config/config";

import { isAuthorisedForGroups } from "@mtfh/common";
import { Asset } from "@mtfh/common/lib/api/asset/v1";
import { Button, Heading, Text } from "@mtfh/common/lib/components";

interface LbhOwnershipInformationProps {
  asset: Asset;
}

export const LbhOwnershipInformation = ({ asset }: LbhOwnershipInformationProps) => {
  let displayMessage;
  if (!asset.assetManagement) {
    displayMessage = "No asset management data found";
  } else if (asset.assetManagement?.isCouncilProperty) {
    displayMessage = locale.lbhOwnershipInformation.ownedByLbh;
  } else {
    displayMessage = locale.lbhOwnershipInformation.notOwnedByLbh;
  }

  return (
    <aside className="mtfh-lbh-ownership-info">
      <Heading
        variant="h2"
        className="lbh-heading lbh-heading-h3"
        data-testid="ownership-info-heading"
      >
        {locale.lbhOwnershipInformation.sideBarHeading}
      </Heading>

      <Text size="sm" data-testid="ownership-info">
        {displayMessage}
      </Text>

      {isAuthorisedForGroups(assetAdminAuthGroups) && (
        <Button
          as={RouterLink}
          to={`/property/${asset.id}/edit-asset-ownership`}
          data-testid="edit-ownership-info-button"
        >
          {locale.lbhOwnershipInformation.editOwnershipButton}
        </Button>
      )}

      <hr className="lbh-horizontal-bar" />
    </aside>
  );
};
