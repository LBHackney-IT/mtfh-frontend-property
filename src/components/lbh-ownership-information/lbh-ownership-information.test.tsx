import React from "react";

import { render } from "@hackney/mtfh-test-utils";
import { screen } from "@testing-library/react";

import { locale } from "../../services";
import { LbhOwnershipInformation } from "./lbh-ownership-information";

import { Asset } from "@mtfh/common/lib/api/asset/v1";
import * as auth from "@mtfh/common/lib/auth/auth";

const assetData: Asset = {
  id: "769894bd-b0bc-47eb-a780-322372c2448f",
  assetId: "0019062023",
  assetType: "Block",
  assetLocation: {
    parentAssets: [],
    floorNo: "0",
    totalBlockFloors: 3,
  },
  assetAddress: {
    uprn: "10008355547",
    addressLine1: "123 Test Block(CAN BE DELETED)",
    addressLine2: "",
    addressLine3: "",
    addressLine4: "",
    postPreamble: "",
    postCode: "FK8 1FH",
  },
  assetManagement: {
    isCouncilProperty: true,
    managingOrganisation: "London Borough of Hackney",
    managingOrganisationId: "c01e3146-e630-c2cd-e709-18ef57bf3724",
    isTMOManaged: false,
    agent: "Agent Smith",
    areaOfficeName: "Test",
    owner: "Test",
  },
  assetCharacteristics: {
    numberOfBedrooms: 0,
    numberOfLifts: 0,
    numberOfLivingRooms: 0,
    windowType: "Wood",
    yearConstructed: "1991",
    numberOfSingleBeds: null,
    numberOfDoubleBeds: null,
    numberOfFloors: null,
    totalBlockFloors: null,
    heating: null,
    propertyFactor: null,
    architecturalType: null,
  },
  versionNumber: 0,
  tenure: {
    id: "c01e3146-e630-c2cd-e709-18ef57bf3724",
    paymentReference: "",
    type: "",
    startOfTenureDate: "",
    endOfTenureDate: "",
    isActive: false,
  },
  rootAsset: "",
  parentAssetIds: "",
  rentGroup: null,
  boilerHouseId: "",
};

beforeEach(() => {
  jest.resetAllMocks();
  jest.spyOn(auth, "isAuthorisedForGroups").mockReturnValue(true);
});

test("it renders the component", async () => {
  const { container } = render(<LbhOwnershipInformation asset={assetData} />);

  await screen.findByText(locale.lbhOwnershipInformation.sideBarHeading);

  expect(container).toMatchSnapshot();
});

test("it says the asset is owned by Hackney Council when 'isCouncilProperty' is true", async () => {
  render(<LbhOwnershipInformation asset={assetData} />);

  const assetOwnership = screen.getByText(locale.lbhOwnershipInformation.ownedByLbh);

  expect(assetOwnership).toBeVisible();
});

test("it says the asset is NOT owned by Hackney Council when 'isCouncilProperty' is false", async () => {
  // Set ownership to false
  assetData.assetManagement.isCouncilProperty = false;

  render(<LbhOwnershipInformation asset={assetData} />);

  const assetOwnership = screen.getByText(locale.lbhOwnershipInformation.notOwnedByLbh);

  expect(assetOwnership).toBeVisible();
});

test("it shows the 'Edit Ownership' button when the user is authorized", async () => {
  render(<LbhOwnershipInformation asset={assetData} />);

  // The button is actually a 'link' element and not a 'button'
  const editOwnershipButton = screen.getByRole("link", {
    name: locale.lbhOwnershipInformation.editOwnershipButton,
  });

  expect(editOwnershipButton).toBeVisible();
});

test("it does NOT the 'Edit Ownership' button when the user is not authorized", async () => {
  jest.spyOn(auth, "isAuthorisedForGroups").mockReturnValue(false);

  render(<LbhOwnershipInformation asset={assetData} />);

  // The button is actually a 'link' element and not a 'button'
  const editOwnershipButton = screen.queryByRole("link", {
    name: locale.lbhOwnershipInformation.editOwnershipButton,
  });

  expect(editOwnershipButton).toBeNull();
});
