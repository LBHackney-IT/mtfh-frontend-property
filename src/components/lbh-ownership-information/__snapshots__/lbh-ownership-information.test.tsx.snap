// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders the component 1`] = `
<div>
  <aside
    class="mtfh-lbh-ownership-info"
  >
    <h2
      class="lbh-heading-h2 lbh-heading lbh-heading-h3"
    >
      Asset ownership
    </h2>
    <p
      class="lbh-body-s"
    >
      This asset is owned by Hackney Council
    </p>
    <a
      class="govuk-button lbh-button"
      href="/property/769894bd-b0bc-47eb-a780-322372c2448f/edit-asset-ownership"
    >
      Edit ownership
    </a>
    <hr
      class="lbh-horizontal-bar"
    />
  </aside>
</div>
`;
